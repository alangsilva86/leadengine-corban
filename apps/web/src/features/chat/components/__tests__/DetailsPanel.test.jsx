/** @vitest-environment jsdom */
import '@testing-library/jest-dom/vitest';
import { cleanup, render, screen } from '@testing-library/react';
import { afterEach, describe, expect, it, vi } from 'vitest';
import { forwardRef } from 'react';

vi.mock('../DetailsPanel/LeadSummaryCard.jsx', () => ({
  __esModule: true,
  default: ({ lead }) => (
    <div data-testid="lead-summary">Lead summary: {lead?.status ?? 'unknown'}</div>
  ),
}));

vi.mock('../DetailsPanel/LeadDetailsTabs.jsx', () => ({
  __esModule: true,
  default: ({ ticket }) => (
    <div data-testid="lead-details">Lead details tabs: {ticket?.id ?? 'no-ticket'}</div>
  ),
}));

vi.mock('../DetailsPanel/ConsentInfo.jsx', () => ({
  __esModule: true,
  default: ({ consent }) => <div data-testid="consent-info">Consent: {JSON.stringify(consent)}</div>,
}));

vi.mock('../DetailsPanel/ProposalMiniSim.jsx', () => ({
  __esModule: true,
  default: () => <div data-testid="proposal-mini-sim">Proposal mini sim</div>,
}));

vi.mock('../DetailsPanel/NotesSection.jsx', () => ({
  __esModule: true,
  default: forwardRef((props, ref) => (
    <div ref={ref} data-testid="notes-section">Notes section</div>
  )),
}));

vi.mock('../DetailsPanel/TasksSection.jsx', () => ({
  __esModule: true,
  default: ({ ticket }) => (
    <div data-testid="tasks-section">Tasks for {ticket?.id ?? 'no-ticket'}</div>
  ),
}));

vi.mock('../DetailsPanel/AuditTrailLink.jsx', () => ({
  __esModule: true,
  default: ({ onOpenAudit }) => (
    <button type="button" data-testid="audit-trail" onClick={onOpenAudit}>
      Abrir auditoria
    </button>
  ),
}));

vi.mock('../Shared/AttachmentPreview.jsx', () => ({
  __esModule: true,
  default: ({ attachments }) => (
    <div data-testid="attachment-preview">Attachments: {attachments.length}</div>
  ),
}));

vi.mock('../Shared/StatusBadge.jsx', () => ({
  __esModule: true,
  default: ({ status }) => <div data-testid="status-badge">Status: {status}</div>,
}));

vi.mock('@/features/contacts/components/ContactSummary.jsx', () => ({
  __esModule: true,
  default: ({ contact }) => (
    <div data-testid="contact-summary">Contact: {contact?.name ?? 'sem nome'}</div>
  ),
}));

import { DetailsPanel } from '../DetailsPanel/DetailsPanel.jsx';

describe('DetailsPanel', () => {
  afterEach(() => {
    cleanup();
    vi.clearAllMocks();
  });

  it('renderiza os grupos de seções previstos em snapshot', () => {
    const ticket = {
      id: 'ticket-1',
      contact: {
        id: 'contact-1',
        name: 'Maria Silva',
        company: 'Empresa X',
        phone: '+55 11 99999-9999',
        email: 'maria@example.com',
        document: '123.456.789-00',
        consent: {
          whatsapp: true,
        },
      },
      lead: {
        status: 'open',
        value: 5000,
      },
      timeline: {
        firstInboundAt: '2023-09-10T10:00:00Z',
        unreadInboundCount: 2,
      },
      metadata: {
        attachments: [
          { id: 'file-1' },
          { id: 'file-2' },
        ],
      },
      notes: [
        { id: 'note-1', body: 'Primeira nota' },
        { id: 'note-2', body: 'Segunda nota' },
      ],
    };

    render(
      <DetailsPanel
        ticket={ticket}
        onCreateNote={vi.fn()}
        notesLoading={false}
        onReopenWindow={vi.fn()}
        onOpenAudit={vi.fn()}
      />
    );

    expect(screen.getByText('Contato principal')).toBeInTheDocument();
    expect(screen.getByText('Proposta rápida')).toBeInTheDocument();
    expect(screen.getByText('Timeline resumida')).toBeInTheDocument();
  });

  it('exibe links de atalho para a command bar', () => {
    const ticket = {
      id: 'ticket-2',
      contact: {
        id: 'contact-2',
        name: 'João Souza',
        phone: '+55 21 98888-7777',
      },
    };

    render(
      <DetailsPanel
        ticket={ticket}
        onCreateNote={vi.fn()}
        notesLoading={false}
        onReopenWindow={vi.fn()}
        onOpenAudit={vi.fn()}
      />,
    );

    expect(screen.getByRole('link', { name: 'Atribuir' })).toBeInTheDocument();
    expect(screen.getByRole('link', { name: 'Telefonia' })).toBeInTheDocument();
    expect(screen.getAllByRole('link', { name: 'Editar contato' })[0]).toBeInTheDocument();
  });
});
