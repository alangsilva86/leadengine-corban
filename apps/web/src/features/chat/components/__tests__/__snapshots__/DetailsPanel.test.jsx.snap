// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`DetailsPanel > renderiza os grupos de seções previstos em snapshot 1`] = `
<div
  class="flex w-full flex-1 min-w-0 flex-col gap-5 overflow-y-auto overflow-x-hidden p-4"
>
  <section
    class="w-full rounded-3xl border border-surface-overlay-glass-border bg-surface-overlay-quiet/60 p-5 shadow-[0_24px_45px_-32px_rgba(15,23,42,0.8)] backdrop-blur"
  >
    <div
      class="flex min-w-0 flex-col gap-4 sm:flex-row sm:items-start sm:justify-between"
    >
      <div
        class="flex flex-col gap-2"
      >
        <span
          class="text-xs font-semibold uppercase tracking-wide text-foreground-muted"
        >
          Contato principal
        </span>
        <h2
          class="text-xl font-semibold leading-tight text-foreground"
        >
          Maria Silva
        </h2>
        <p
          class="text-sm text-foreground-muted"
        >
          Empresa X
        </p>
      </div>
      <div
        data-testid="status-badge"
      >
        Status: 
        open
      </div>
    </div>
    <dl
      class="mt-5 grid min-w-0 gap-4 md:grid-cols-2"
    >
      <div
        class="flex min-w-0 flex-col gap-1"
      >
        <span
          class="text-xs font-semibold uppercase tracking-wide text-foreground-muted"
        >
          Telefone
        </span>
        <div
          class="text-sm text-foreground flex flex-wrap items-center gap-2 min-w-0"
        >
          <div
            class="flex min-w-0 items-center gap-2 break-words"
          >
            <svg
              aria-hidden="true"
              class="lucide lucide-phone text-foreground-muted size-4 shrink-0"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M13.832 16.568a1 1 0 0 0 1.213-.303l.355-.465A2 2 0 0 1 17 15h3a2 2 0 0 1 2 2v3a2 2 0 0 1-2 2A18 18 0 0 1 2 4a2 2 0 0 1 2-2h3a2 2 0 0 1 2 2v3a2 2 0 0 1-.8 1.6l-.468.351a1 1 0 0 0-.292 1.233 14 14 0 0 0 6.392 6.384"
              />
            </svg>
            <span
              class="font-medium break-words break-all max-w-full"
            >
              +55 11 99999-9999
            </span>
          </div>
          <button
            aria-label="Copiar telefone"
            class="inline-flex items-center justify-center whitespace-nowrap rounded-[var(--radius)] text-sm font-medium transition-all disabled:pointer-events-none disabled:opacity-60 [&_svg]:pointer-events-none [&_svg:not([class*='size-'])]:size-4 [&_svg]:shrink-0 outline-none focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-[color:color-mix(in_srgb,var(--ring)_75%,transparent)] focus-visible:ring-offset-2 focus-visible:ring-offset-[color:color-mix(in_srgb,var(--bg)_92%,transparent)] hover:bg-[color-mix(in_oklab,_var(--surface)_18%,_transparent)] min-h-[38px] gap-1.5 px-3 has-[>svg]:px-2.5 shrink-0 text-foreground-muted hover:text-foreground"
            data-slot="button"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="lucide lucide-copy size-4"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <rect
                height="14"
                rx="2"
                ry="2"
                width="14"
                x="8"
                y="8"
              />
              <path
                d="M4 16c-1.1 0-2-.9-2-2V4c0-1.1.9-2 2-2h10c1.1 0 2 .9 2 2"
              />
            </svg>
          </button>
        </div>
      </div>
      <div
        class="flex min-w-0 flex-col gap-1"
      >
        <span
          class="text-xs font-semibold uppercase tracking-wide text-foreground-muted"
        >
          E-mail
        </span>
        <div
          class="text-sm text-foreground flex flex-wrap items-center gap-2 min-w-0"
        >
          <div
            class="flex min-w-0 items-center gap-2 break-words"
          >
            <svg
              aria-hidden="true"
              class="lucide lucide-mail text-foreground-muted size-4 shrink-0"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="m22 7-8.991 5.727a2 2 0 0 1-2.009 0L2 7"
              />
              <rect
                height="16"
                rx="2"
                width="20"
                x="2"
                y="4"
              />
            </svg>
            <span
              class="font-medium break-words break-all max-w-full"
            >
              maria@example.com
            </span>
          </div>
          <button
            aria-label="Copiar e-mail"
            class="inline-flex items-center justify-center whitespace-nowrap rounded-[var(--radius)] text-sm font-medium transition-all disabled:pointer-events-none disabled:opacity-60 [&_svg]:pointer-events-none [&_svg:not([class*='size-'])]:size-4 [&_svg]:shrink-0 outline-none focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-[color:color-mix(in_srgb,var(--ring)_75%,transparent)] focus-visible:ring-offset-2 focus-visible:ring-offset-[color:color-mix(in_srgb,var(--bg)_92%,transparent)] hover:bg-[color-mix(in_oklab,_var(--surface)_18%,_transparent)] min-h-[38px] gap-1.5 px-3 has-[>svg]:px-2.5 shrink-0 text-foreground-muted hover:text-foreground"
            data-slot="button"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="lucide lucide-copy size-4"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <rect
                height="14"
                rx="2"
                ry="2"
                width="14"
                x="8"
                y="8"
              />
              <path
                d="M4 16c-1.1 0-2-.9-2-2V4c0-1.1.9-2 2-2h10c1.1 0 2 .9 2 2"
              />
            </svg>
          </button>
        </div>
      </div>
      <div
        class="flex min-w-0 flex-col gap-1"
      >
        <span
          class="text-xs font-semibold uppercase tracking-wide text-foreground-muted"
        >
          Documento
        </span>
        <div
          class="text-sm text-foreground flex flex-wrap items-center gap-2 min-w-0"
        >
          <div
            class="flex min-w-0 items-center gap-2 break-words"
          >
            <svg
              aria-hidden="true"
              class="lucide lucide-file-text text-foreground-muted size-4 shrink-0"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M15 2H6a2 2 0 0 0-2 2v16a2 2 0 0 0 2 2h12a2 2 0 0 0 2-2V7Z"
              />
              <path
                d="M14 2v4a2 2 0 0 0 2 2h4"
              />
              <path
                d="M10 9H8"
              />
              <path
                d="M16 13H8"
              />
              <path
                d="M16 17H8"
              />
            </svg>
            <span
              class="font-medium break-words break-all max-w-full"
            >
              123.456.789-00
            </span>
          </div>
          <button
            aria-label="Copiar documento"
            class="inline-flex items-center justify-center whitespace-nowrap rounded-[var(--radius)] text-sm font-medium transition-all disabled:pointer-events-none disabled:opacity-60 [&_svg]:pointer-events-none [&_svg:not([class*='size-'])]:size-4 [&_svg]:shrink-0 outline-none focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-[color:color-mix(in_srgb,var(--ring)_75%,transparent)] focus-visible:ring-offset-2 focus-visible:ring-offset-[color:color-mix(in_srgb,var(--bg)_92%,transparent)] hover:bg-[color-mix(in_oklab,_var(--surface)_18%,_transparent)] min-h-[38px] gap-1.5 px-3 has-[>svg]:px-2.5 shrink-0 text-foreground-muted hover:text-foreground"
            data-slot="button"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="lucide lucide-copy size-4"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <rect
                height="14"
                rx="2"
                ry="2"
                width="14"
                x="8"
                y="8"
              />
              <path
                d="M4 16c-1.1 0-2-.9-2-2V4c0-1.1.9-2 2-2h10c1.1 0 2 .9 2 2"
              />
            </svg>
          </button>
        </div>
      </div>
      <div
        class="flex min-w-0 flex-col gap-1"
      >
        <span
          class="text-xs font-semibold uppercase tracking-wide text-foreground-muted"
        >
          ID do contato
        </span>
        <div
          class="text-sm text-foreground flex flex-wrap items-center gap-2 min-w-0"
        >
          <div
            class="flex min-w-0 items-center gap-2 break-words"
          >
            <svg
              aria-hidden="true"
              class="lucide lucide-info text-foreground-muted size-4 shrink-0"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <circle
                cx="12"
                cy="12"
                r="10"
              />
              <path
                d="M12 16v-4"
              />
              <path
                d="M12 8h.01"
              />
            </svg>
            <span
              class="font-medium break-words break-all max-w-full"
            >
              contact-1
            </span>
          </div>
          <button
            aria-label="Copiar id do contato"
            class="inline-flex items-center justify-center whitespace-nowrap rounded-[var(--radius)] text-sm font-medium transition-all disabled:pointer-events-none disabled:opacity-60 [&_svg]:pointer-events-none [&_svg:not([class*='size-'])]:size-4 [&_svg]:shrink-0 outline-none focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-[color:color-mix(in_srgb,var(--ring)_75%,transparent)] focus-visible:ring-offset-2 focus-visible:ring-offset-[color:color-mix(in_srgb,var(--bg)_92%,transparent)] hover:bg-[color-mix(in_oklab,_var(--surface)_18%,_transparent)] min-h-[38px] gap-1.5 px-3 has-[>svg]:px-2.5 shrink-0 text-foreground-muted hover:text-foreground"
            data-slot="button"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="lucide lucide-copy size-4"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <rect
                height="14"
                rx="2"
                ry="2"
                width="14"
                x="8"
                y="8"
              />
              <path
                d="M4 16c-1.1 0-2-.9-2-2V4c0-1.1.9-2 2-2h10c1.1 0 2 .9 2 2"
              />
            </svg>
          </button>
        </div>
      </div>
    </dl>
  </section>
  <div
    class="flex flex-1 min-w-0 flex-col gap-5"
    data-orientation="horizontal"
    data-slot="tabs"
    dir="ltr"
  >
    <div
      aria-orientation="horizontal"
      class="text-muted-foreground h-9 rounded-lg flex w-full min-w-0 flex-wrap items-center justify-start gap-2 overflow-x-auto bg-surface-overlay-quiet/60 p-1.5 md:flex-nowrap"
      data-orientation="horizontal"
      data-slot="tabs-list"
      role="tablist"
      style="outline: none;"
      tabindex="0"
    >
      <button
        aria-controls="radix-_r_0_-content-contact"
        aria-selected="true"
        class="data-[state=active]:bg-background dark:data-[state=active]:text-foreground focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:outline-ring dark:data-[state=active]:border-input dark:data-[state=active]:bg-input/30 text-foreground dark:text-muted-foreground inline-flex h-[calc(100%-1px)] flex-1 items-center justify-center gap-1.5 rounded-md border border-transparent px-2 py-1 text-sm font-medium whitespace-nowrap transition-[color,box-shadow] focus-visible:ring-[3px] focus-visible:outline-1 disabled:pointer-events-none disabled:opacity-50 data-[state=active]:shadow-sm [&_svg]:pointer-events-none [&_svg]:shrink-0 [&_svg:not([class*='size-'])]:size-4"
        data-orientation="horizontal"
        data-radix-collection-item=""
        data-slot="tabs-trigger"
        data-state="active"
        id="radix-_r_0_-trigger-contact"
        role="tab"
        tabindex="-1"
        type="button"
      >
        Contato
      </button>
      <button
        aria-controls="radix-_r_0_-content-opportunity"
        aria-selected="false"
        class="data-[state=active]:bg-background dark:data-[state=active]:text-foreground focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:outline-ring dark:data-[state=active]:border-input dark:data-[state=active]:bg-input/30 text-foreground dark:text-muted-foreground inline-flex h-[calc(100%-1px)] flex-1 items-center justify-center gap-1.5 rounded-md border border-transparent px-2 py-1 text-sm font-medium whitespace-nowrap transition-[color,box-shadow] focus-visible:ring-[3px] focus-visible:outline-1 disabled:pointer-events-none disabled:opacity-50 data-[state=active]:shadow-sm [&_svg]:pointer-events-none [&_svg]:shrink-0 [&_svg:not([class*='size-'])]:size-4"
        data-orientation="horizontal"
        data-radix-collection-item=""
        data-slot="tabs-trigger"
        data-state="inactive"
        id="radix-_r_0_-trigger-opportunity"
        role="tab"
        tabindex="-1"
        type="button"
      >
        Oportunidade
      </button>
      <button
        aria-controls="radix-_r_0_-content-timeline"
        aria-selected="false"
        class="data-[state=active]:bg-background dark:data-[state=active]:text-foreground focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:outline-ring dark:data-[state=active]:border-input dark:data-[state=active]:bg-input/30 text-foreground dark:text-muted-foreground inline-flex h-[calc(100%-1px)] flex-1 items-center justify-center gap-1.5 rounded-md border border-transparent px-2 py-1 text-sm font-medium whitespace-nowrap transition-[color,box-shadow] focus-visible:ring-[3px] focus-visible:outline-1 disabled:pointer-events-none disabled:opacity-50 data-[state=active]:shadow-sm [&_svg]:pointer-events-none [&_svg]:shrink-0 [&_svg:not([class*='size-'])]:size-4"
        data-orientation="horizontal"
        data-radix-collection-item=""
        data-slot="tabs-trigger"
        data-state="inactive"
        id="radix-_r_0_-trigger-timeline"
        role="tab"
        tabindex="-1"
        type="button"
      >
        Timeline
      </button>
      <button
        aria-controls="radix-_r_0_-content-attachments"
        aria-selected="false"
        class="data-[state=active]:bg-background dark:data-[state=active]:text-foreground focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:outline-ring dark:data-[state=active]:border-input dark:data-[state=active]:bg-input/30 text-foreground dark:text-muted-foreground inline-flex h-[calc(100%-1px)] flex-1 items-center justify-center gap-1.5 rounded-md border border-transparent px-2 py-1 text-sm font-medium whitespace-nowrap transition-[color,box-shadow] focus-visible:ring-[3px] focus-visible:outline-1 disabled:pointer-events-none disabled:opacity-50 data-[state=active]:shadow-sm [&_svg]:pointer-events-none [&_svg]:shrink-0 [&_svg:not([class*='size-'])]:size-4"
        data-orientation="horizontal"
        data-radix-collection-item=""
        data-slot="tabs-trigger"
        data-state="inactive"
        id="radix-_r_0_-trigger-attachments"
        role="tab"
        tabindex="-1"
        type="button"
      >
        Anexos & Notas
      </button>
    </div>
    <div
      aria-labelledby="radix-_r_0_-trigger-contact"
      class="flex-1 outline-none space-y-4 min-w-0"
      data-orientation="horizontal"
      data-slot="tabs-content"
      data-state="active"
      id="radix-_r_0_-content-contact"
      role="tabpanel"
      style="animation-duration: 0s;"
      tabindex="0"
    >
      <div
        class="space-y-3 w-full min-w-0"
        data-orientation="vertical"
        data-slot="accordion"
      >
        <div
          class="last:border-b-0 w-full min-w-0 overflow-hidden rounded-2xl border border-surface-overlay-glass-border bg-surface-overlay-quiet/40 px-2 backdrop-blur"
          data-orientation="vertical"
          data-slot="accordion-item"
          data-state="open"
        >
          <h3
            class="flex"
            data-orientation="vertical"
            data-state="open"
          >
            <button
              aria-controls="radix-_r_6_"
              aria-expanded="true"
              class="focus-visible:border-ring transition-all outline-none focus-visible:ring-[3px] disabled:pointer-events-none disabled:opacity-50 [&[data-state=open]>svg]:rotate-180 hover:no-underline focus-visible:ring-ring/50 flex w-full flex-1 flex-wrap items-start justify-between gap-4 rounded-xl px-3 py-4 text-left text-sm font-semibold text-foreground"
              data-orientation="vertical"
              data-radix-collection-item=""
              data-slot="accordion-trigger"
              data-state="open"
              id="radix-_r_5_"
              type="button"
            >
              <div
                class="flex min-w-0 flex-1 items-start gap-3"
              >
                <span
                  class="bg-primary/10 text-primary flex size-10 items-center justify-center rounded-full"
                >
                  <svg
                    aria-hidden="true"
                    class="lucide lucide-circle-user-round size-4"
                    fill="none"
                    height="24"
                    stroke="currentColor"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="2"
                    viewBox="0 0 24 24"
                    width="24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M18 20a6 6 0 0 0-12 0"
                    />
                    <circle
                      cx="12"
                      cy="10"
                      r="4"
                    />
                    <circle
                      cx="12"
                      cy="12"
                      r="10"
                    />
                  </svg>
                </span>
                <div
                  class="flex min-w-0 flex-1 flex-col gap-1"
                >
                  <div
                    class="flex min-w-0 flex-wrap items-center gap-2"
                  >
                    <span
                      class="text-sm font-semibold"
                    >
                      Ficha do contato
                    </span>
                  </div>
                  <p
                    class="text-xs font-normal text-foreground-muted break-words"
                  >
                    Informações cadastrais e campos personalizados organizadas por categoria.
                  </p>
                </div>
              </div>
              <svg
                aria-hidden="true"
                class="lucide lucide-chevron-down text-muted-foreground pointer-events-none size-4 shrink-0 translate-y-0.5 transition-transform duration-200"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="m6 9 6 6 6-6"
                />
              </svg>
            </button>
          </h3>
          <div
            aria-labelledby="radix-_r_5_"
            class="data-[state=closed]:animate-accordion-up data-[state=open]:animate-accordion-down overflow-hidden text-sm"
            data-orientation="vertical"
            data-slot="accordion-content"
            data-state="open"
            id="radix-_r_6_"
            role="region"
            style="--radix-accordion-content-height: var(--radix-collapsible-content-height); --radix-accordion-content-width: var(--radix-collapsible-content-width); transition-duration: 0s; animation-name: none;"
          >
            <div
              class="pt-0 pb-4 px-0"
            >
              <div
                class="min-w-0 max-w-full overflow-hidden rounded-xl border border-surface-overlay-glass-border bg-surface-overlay-quiet/80 p-4 text-sm text-foreground"
              >
                <div
                  data-testid="contact-summary"
                >
                  Contact: 
                  Maria Silva
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="last:border-b-0 w-full min-w-0 overflow-hidden rounded-2xl border border-surface-overlay-glass-border bg-surface-overlay-quiet/40 px-2 backdrop-blur"
          data-orientation="vertical"
          data-slot="accordion-item"
          data-state="open"
        >
          <h3
            class="flex"
            data-orientation="vertical"
            data-state="open"
          >
            <button
              aria-controls="radix-_r_8_"
              aria-expanded="true"
              class="focus-visible:border-ring transition-all outline-none focus-visible:ring-[3px] disabled:pointer-events-none disabled:opacity-50 [&[data-state=open]>svg]:rotate-180 hover:no-underline focus-visible:ring-ring/50 flex w-full flex-1 flex-wrap items-start justify-between gap-4 rounded-xl px-3 py-4 text-left text-sm font-semibold text-foreground"
              data-orientation="vertical"
              data-radix-collection-item=""
              data-slot="accordion-trigger"
              data-state="open"
              id="radix-_r_7_"
              type="button"
            >
              <div
                class="flex min-w-0 flex-1 items-start gap-3"
              >
                <span
                  class="bg-primary/10 text-primary flex size-10 items-center justify-center rounded-full"
                >
                  <svg
                    aria-hidden="true"
                    class="lucide lucide-info size-4"
                    fill="none"
                    height="24"
                    stroke="currentColor"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="2"
                    viewBox="0 0 24 24"
                    width="24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <circle
                      cx="12"
                      cy="12"
                      r="10"
                    />
                    <path
                      d="M12 16v-4"
                    />
                    <path
                      d="M12 8h.01"
                    />
                  </svg>
                </span>
                <div
                  class="flex min-w-0 flex-1 flex-col gap-1"
                >
                  <div
                    class="flex min-w-0 flex-wrap items-center gap-2"
                  >
                    <span
                      class="text-sm font-semibold"
                    >
                      Campos do atendimento
                    </span>
                  </div>
                  <p
                    class="text-xs font-normal text-foreground-muted break-words"
                  >
                    Campos personalizados e dados operacionais do ticket.
                  </p>
                </div>
              </div>
              <svg
                aria-hidden="true"
                class="lucide lucide-chevron-down text-muted-foreground pointer-events-none size-4 shrink-0 translate-y-0.5 transition-transform duration-200"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="m6 9 6 6 6-6"
                />
              </svg>
            </button>
          </h3>
          <div
            aria-labelledby="radix-_r_7_"
            class="data-[state=closed]:animate-accordion-up data-[state=open]:animate-accordion-down overflow-hidden text-sm"
            data-orientation="vertical"
            data-slot="accordion-content"
            data-state="open"
            id="radix-_r_8_"
            role="region"
            style="--radix-accordion-content-height: var(--radix-collapsible-content-height); --radix-accordion-content-width: var(--radix-collapsible-content-width); transition-duration: 0s; animation-name: none;"
          >
            <div
              class="pt-0 pb-4 px-0"
            >
              <div
                class="min-w-0 max-w-full overflow-hidden rounded-xl border border-surface-overlay-glass-border bg-surface-overlay-quiet/80 p-4 text-sm text-foreground"
              >
                <div
                  data-testid="lead-details"
                >
                  Lead details tabs: 
                  ticket-1
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="last:border-b-0 w-full min-w-0 overflow-hidden rounded-2xl border border-surface-overlay-glass-border bg-surface-overlay-quiet/40 px-2 backdrop-blur"
          data-orientation="vertical"
          data-slot="accordion-item"
          data-state="open"
        >
          <h3
            class="flex"
            data-orientation="vertical"
            data-state="open"
          >
            <button
              aria-controls="radix-_r_a_"
              aria-expanded="true"
              class="focus-visible:border-ring transition-all outline-none focus-visible:ring-[3px] disabled:pointer-events-none disabled:opacity-50 [&[data-state=open]>svg]:rotate-180 hover:no-underline focus-visible:ring-ring/50 flex w-full flex-1 flex-wrap items-start justify-between gap-4 rounded-xl px-3 py-4 text-left text-sm font-semibold text-foreground"
              data-orientation="vertical"
              data-radix-collection-item=""
              data-slot="accordion-trigger"
              data-state="open"
              id="radix-_r_9_"
              type="button"
            >
              <div
                class="flex min-w-0 flex-1 items-start gap-3"
              >
                <span
                  class="bg-primary/10 text-primary flex size-10 items-center justify-center rounded-full"
                >
                  <svg
                    aria-hidden="true"
                    class="lucide lucide-shield-check size-4"
                    fill="none"
                    height="24"
                    stroke="currentColor"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="2"
                    viewBox="0 0 24 24"
                    width="24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M20 13c0 5-3.5 7.5-7.66 8.95a1 1 0 0 1-.67-.01C7.5 20.5 4 18 4 13V6a1 1 0 0 1 1-1c2 0 4.5-1.2 6.24-2.72a1.17 1.17 0 0 1 1.52 0C14.51 3.81 17 5 19 5a1 1 0 0 1 1 1z"
                    />
                    <path
                      d="m9 12 2 2 4-4"
                    />
                  </svg>
                </span>
                <div
                  class="flex min-w-0 flex-1 flex-col gap-1"
                >
                  <div
                    class="flex min-w-0 flex-wrap items-center gap-2"
                  >
                    <span
                      class="text-sm font-semibold"
                    >
                      Consentimento
                    </span>
                  </div>
                  <p
                    class="text-xs font-normal text-foreground-muted break-words"
                  >
                    Preferências de comunicação registradas para o contato.
                  </p>
                </div>
              </div>
              <svg
                aria-hidden="true"
                class="lucide lucide-chevron-down text-muted-foreground pointer-events-none size-4 shrink-0 translate-y-0.5 transition-transform duration-200"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="m6 9 6 6 6-6"
                />
              </svg>
            </button>
          </h3>
          <div
            aria-labelledby="radix-_r_9_"
            class="data-[state=closed]:animate-accordion-up data-[state=open]:animate-accordion-down overflow-hidden text-sm"
            data-orientation="vertical"
            data-slot="accordion-content"
            data-state="open"
            id="radix-_r_a_"
            role="region"
            style="--radix-accordion-content-height: var(--radix-collapsible-content-height); --radix-accordion-content-width: var(--radix-collapsible-content-width); transition-duration: 0s; animation-name: none;"
          >
            <div
              class="pt-0 pb-4 px-0"
            >
              <div
                class="min-w-0 max-w-full overflow-hidden rounded-xl border border-surface-overlay-glass-border bg-surface-overlay-quiet/80 p-4 text-sm text-foreground"
              >
                <div
                  data-testid="consent-info"
                >
                  Consent: 
                  {"whatsapp":true}
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      aria-labelledby="radix-_r_0_-trigger-opportunity"
      class="flex-1 outline-none space-y-4 min-w-0"
      data-orientation="horizontal"
      data-slot="tabs-content"
      data-state="inactive"
      hidden=""
      id="radix-_r_0_-content-opportunity"
      role="tabpanel"
      tabindex="0"
    />
    <div
      aria-labelledby="radix-_r_0_-trigger-timeline"
      class="flex-1 outline-none space-y-4 min-w-0"
      data-orientation="horizontal"
      data-slot="tabs-content"
      data-state="inactive"
      hidden=""
      id="radix-_r_0_-content-timeline"
      role="tabpanel"
      tabindex="0"
    />
    <div
      aria-labelledby="radix-_r_0_-trigger-attachments"
      class="flex-1 outline-none space-y-4 min-w-0"
      data-orientation="horizontal"
      data-slot="tabs-content"
      data-state="inactive"
      hidden=""
      id="radix-_r_0_-content-attachments"
      role="tabpanel"
      tabindex="0"
    />
  </div>
</div>
`;
